client/uploadable.go

```diff
@@ -296,14 +306,17 @@ func (u *AutoUploadable) messageHandler(requestID string, msg *protocol.Envelope
 
 	if responseError != nil {
 		status = responseError.Status
-		message = responseError.Message
+		message = responseError
 
 		logger.Errorf("error while executing operation %s: %s", operation, responseError.Message)
 	}
 
+	headers := protocol.NewHeaders(protocol.WithCorrelationID(msg.Headers.CorrelationID()),
+		protocol.WithContentType("application/json"))
+
 	reply := &protocol.Envelope{
 		Topic:   msg.Topic,                                           // preserve the topic
-		Headers: msg.Headers,                                         // preserve the headers
+		Headers: headers,                                             // preserve the headers
```

Comment:
The headers are not preserved any more. The comment is not correct.

Reply:
Okay, I will adjust the comment to suit the new case.